src/utils/user-utils/user-services.js

```diff
@@ -10,13 +17,75 @@ export const getUsers = createAsyncThunk("users/getUsers", async () => {
   } catch (error) {}
 });
 
-export const getUser = createAsyncThunk("users/getUser", async (id) => {
+export const getUser = createAsyncThunk("users/getUser", async ({userId,setIsLoading}) => {
   try {
-    const res = await axios.get(`/api/users/${id}`);
+    const res = await axios.get(`/api/users/${userId}`);
     if (res.status === 200) {
+      setIsLoading(false)
       return res.data.user;
     }
   } catch (error) {
     console.log("error occured", error);
   }
 });
+
+export const followUser=createAsyncThunk("users/followUser",async({_id,setIsFollowLoading})=>{
+  const encodedToken=localStorage.getItem('token')
+  try {
+    const res=await axios({
+      method:"POST",
+      url:`/api/users/follow/${_id}`,
+      headers:{authorization:encodedToken}
+    })
+    if(res.status===200){
+      setIsFollowLoading(false)
```

Comment:
instead of doing this you should properties of create async thunk and set loading like that